# Curso de WEB SCRAPING con PYTHON para PRINCIPIANTES

https://www.youtube.com/watch?v=bK3EwIMHm94

Basicamente el web scraping es la obtencion de una pagina web de manera automatizada

# Paso 1 instalar todas las librerias
- requests 

# Para ver el codigo fuente de una pagina Ctrl + U ó F12

In [1]:
import requests
from bs4 import BeautifulSoup

In [ ]:
url = "https://dockerlabs.es"

respuesta = requests.get(url)

respuesta.encoding = "UTF-8" #Probe esto por que me mostraba caracteres mal puestos, y pense que estaba mal la codificacion, pero no soluciono

In [20]:
print(respuesta.text)

<!DOCTYPE html>
<html lang="es">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DockerLabs</title>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.11.3/font/bootstrap-icons.min.css">
    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Fira+Code:wght@400;700&display=swap">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <meta name="csrf-token" content="463938ad4820525fc3fd03f811f3906314efae961b3bb33f105664d9dbdde654">

    
    <link rel="stylesheet" href="/static/css/styles.css">
    

    <link rel="icon" href="/static/images/logos/logo.png" type="image/png">

    <script nonce="u4VtCB0haYcGt3Q5udXsIQ">
        window.dataLayer = window.dataLayer || [];
        function gtag() { dataLayer.push(arguments); }
        gtag('js', new Date());
        gtag('config', 'G-ZBED41BLPR');
    </script>

   

In [12]:
if respuesta.status_code == 200:
    soup = BeautifulSoup(respuesta.text, 'html.parser')

    # Con el filtro soup que creamos vamos a usar el metodo find_all, para filtrar tambien aquellos divs
    # que tengan una llamada a la accion onclick

    maquinas = soup.find_all('div', onclick = True)

    #Ahora queremos extraer el texto de estos divs, entonces vamos a usar un bucle for para recorrer la informacion

    for maquina in maquinas:
        
        onclick_text = maquina['onclick']  #para extraer la informacion que esta proxima al elemento onclick
        
        nombre_maquina = onclick_text.split('"')[1] #para poder obtener el nombre de cada maquina que esta entre las comillas simples

        print(nombre_maquina)
else:
    print(f"Hubo un error al hacer la peticion {respuesta.status_code}")

Jason
SocialHub
Psycho
Dance-Samba
Peque\u00f1as-Mentirosas
Veneno
Injection
Grandma
Apolos
Report
Swiss
WhereIsMyWebShell
Inclusion
Collections
Trust
Crackoff
Predictable
BreakMySSH
NodeClimb
Library
ConsoleLog
HackZones
PingPong
FirstHacking
Reverse
MyBB
Hidden
404-not-found
0xc0ffee
DebugMe
Stranger
Stack
UserSearch
Vulnerame
chmod-4755
Lfi.elf
Candy
Verdejo
HedgeHog
BorazuwarahCTF
Sender
Insecure
Upload
Domain
DevTools
Move
Database
Vacaciones
Dark
Rubiks
Forgotten_Portal
Force
SecretJenkins
HiddenCat
Fileception
HackMeDaddy
Backend
Vulnvault
Allien
HereBash
Subversion (Beta)
Reflection
BadPlugin
Raas
Extraviado
cineHack
Seeker
Tproot
Internship
Canario
ApiBase
Gitea
secureLAB
StackInferno
ChocoPing
Bicho
BaluFood
LogisticCloud
galeria
Bypassme
ApacheByte
Minecraft
Status
Ejotapete
Grooti
Waffy
Aidor
Memesploit
WalkingCMS
Mirame
Rutas
NorC
ChatMe
ShowTime
Pinguinazo
Balulero
Inj3ct0rss
ChocolateLovers
Sites
Redirection
r00tless
-Pn
BruteShock
Escolares
AnonymousPingu
File
HackPengu

# Si tambien quiero saber la cantidad de maquinas, puedo agregar un contador dentro del bucle

In [13]:
if respuesta.status_code == 200:
    soup = BeautifulSoup(respuesta.text, 'html.parser')

    # Con el filtro soup que creamos vamos a usar el metodo find_all, para filtrar tambien aquellos divs
    # que tengan una llamada a la accion onclick

    maquinas = soup.find_all('div', onclick = True)
    conteo_maquinas = 1
    #Ahora queremos extraer el texto de estos divs, entonces vamos a usar un bucle for para recorrer la informacion

    for maquina in maquinas:
        
        onclick_text = maquina['onclick']  #para extraer la informacion que esta proxima al elemento onclick
        
        nombre_maquina = onclick_text.split('"')[1] #para poder obtener el nombre de cada maquina que esta entre las comillas simples

        conteo_maquinas +=1

    print(f"El numero de maquinas encontradas es: {conteo_maquinas}")
    
else:
    print(f"Hubo un error al hacer la peticion {respuesta.status_code}")

El numero de maquinas encontradas es: 177


Una vez que ya sabemos la cantidad de maquinas actuales, podriamos hacer una condicion de que si (el conteo_maquinas > 177)
quieero que agregue 1 al contador y me diga, hay una maquina nueva, y mostramos el nuevo nombre de la maquina

# Ahora vamos a impirimir el nombre de cada maquina, el autor de cada maquina, y su dificultad

In [26]:
if respuesta.status_code == 200:
    soup = BeautifulSoup(respuesta.text, 'html.parser')

    # Con el filtro soup que creamos vamos a usar el metodo find_all, para filtrar tambien aquellos divs
    # que tengan una llamada a la accion onclick

    maquinas = soup.find_all('div', onclick = True)
    conteo_maquinas = 1
    #Ahora queremos extraer el texto de estos divs, entonces vamos a usar un bucle for para recorrer la informacion

    autores = set() #conjunto para que no hayan datos repetidos

    for maquina in maquinas:
        onclick_text = maquina['onclick']
        nombre = onclick_text.split('"')[1]

        dificultad = onclick_text.split('"')[3]

        autor = onclick_text.split('"')[7]

        print(f"Nombre: {nombre} -- > Dificultad: {dificultad} --> Autor: {autor}")
    
else:
    print(f"Hubo un error al hacer la peticion {respuesta.status_code}")

Nombre: Jason -- > Dificultad: Medio --> Autor: El Ping\u00fcino de Mario
Nombre: SocialHub -- > Dificultad: Dif\u00edcil --> Autor: El Ping\u00fcino de Mario
Nombre: Psycho -- > Dificultad: F\u00e1cil --> Autor: Luisillo_o
Nombre: Dance-Samba -- > Dificultad: Medio --> Autor: d1se0
Nombre: Peque\u00f1as-Mentirosas -- > Dificultad: F\u00e1cil --> Autor: beafn28
Nombre: Veneno -- > Dificultad: Medio --> Autor: firstatack
Nombre: Injection -- > Dificultad: F\u00e1cil --> Autor: El Ping\u00fcino de Mario
Nombre: Grandma -- > Dificultad: Dif\u00edcil --> Autor: Pylon
Nombre: Apolos -- > Dificultad: Medio --> Autor: Luisillo_o
Nombre: Report -- > Dificultad: Medio --> Autor: TLuisillo_o
Nombre: Swiss -- > Dificultad: Medio --> Autor: darksblack
Nombre: WhereIsMyWebShell -- > Dificultad: F\u00e1cil --> Autor: El Ping\u00fcino de Mario
Nombre: Inclusion -- > Dificultad: Medio --> Autor: El Ping\u00fcino de Mario
Nombre: Collections -- > Dificultad: Medio --> Autor: El Ping\u00fcino de Mario
N